src/components/CountrySelector/index.js

```diff
@@ -44,16 +44,18 @@ const getCountryOptionTemplate = (country: CountryData) =>
     <span class="${style.countryLabel}">${country.name}</span>`
 
 class CountrySelection extends Component<Props, State> {
+  autocompleteEl: ?HTMLDivElement
   state = {
     showNoResultsError: false,
   }
 
-  handleCountrySearchConfirm = (selectedCountry: CountryData) => {
+  handleCountrySearchConfirm = async (selectedCountry: CountryData) => {
     if (selectedCountry) {
       this.setState({
         showNoResultsError: false,
       })
-      this.props.actions.setIdDocumentIssuingCountry(selectedCountry)
+      await this.props.actions.setIdDocumentIssuingCountry(selectedCountry)
```

Comment:
Why you're turning this action call into async? I don't see any network request here? Unless you want to defer this call to next process tick, then I think it should be triggered via `setTimeout(handler, 0)`

Reply:
Can confirm there's no network requests here for setting the issuing country.